File: app/src/main/java/com/ragdroid/dahaka/app/AppDataBindingComponent.java
package com.ragdroid.dahaka.app;

import android.app.Application;
import android.databinding.DataBindingComponent;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;

/**
 * Created just for the purpose of injecting the {@code RequestManager}
 * Created by garimajain on 20/08/17.
 */
@Singleton
@Component(modules = AppDataBindingModule.class)
public interface AppDataBindingComponent extends DataBindingComponent  {

    @Component.Builder
    interface Builder {

        AppDataBindingComponent build();

        @BindsInstance
        Builder application(Application application);
    }
}

File: app/src/main/java/com/ragdroid/dahaka/app/AppDataBindingComponent.kt
package com.ragdroid.dahaka.app

import com.ragdroid.dahaka.util.BindingAdapter
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by garimajain on 10/04/18.
 */
@Singleton
class AppDataBindingComponent @Inject
constructor(private val bindingAdapter: BindingAdapter) : android.databinding.DataBindingComponent {
    override fun getBindingAdapter(): BindingAdapter {
        return bindingAdapter
    }
}